app/helpers/component-argument.js

```diff
@@ -0,0 +1,25 @@
+import { helper } from '@ember/component/helper';
+import { isArray } from '@ember/array';
+
+/**
+ * Determine the type of the component argument for display
+ *
+ * @method componentArgumentDisplay
+ * @param {*} argument
+ * @return {*} The argument with the correct type for display
+ */
+export function componentArgumentDisplay([argument]) {
+  if (typeof argument === 'string') {
+    return `"${argument}"`;
+  } else if (argument === null) {
+    return 'null';
+  } else if (argument === undefined) {
+    return 'undefined';
+  } else if(isArray(argument) || typeof argument === 'object'){
+    return '...';
+  }
+
+  return argument;
```

Comment:
There are three main cases we need to handle 1) when it's a string, we need to quote with `JSON.stringify` (to handle quotes inside the string), 2) when it's typeof object and not `null`, we need to `...` (make this a link to the object inspector later), 3) for everything else, `String(...)` should do the trick (`null`, `undefined`, `true`, `false`, numbers).

The serialization guarantees that it's either a primitive (`null`, `undefined`, `true`, `false`, numbers, strings) or an object like `{ id: <object inspector id> }`. See `_serializeItem` in `render-tree.js`.

Reply:
This is still needed. It fixes the problem with the string serialization (when the string has quotes, etc), and remove the unneeded branches.

```js
if (typeof argument === 'string') {
  return JSON.stringify(argument);
} else if (typeof argument === 'object' && argument !== null) {
  return '...';
} else {
  return String(argument);
}
```